app/modules/sightings/models.py

```diff
@@ -30,6 +31,15 @@ class Sighting(db.Model, FeatherModel):
 
     assets = db.relationship('SightingAssets')
 
+    # Trying to insert enums into DBs seems to cause all sorts of migration problems
+    class Stage(str, enum.Enum):
+        identification = 'identification'
+        un_reviewed = 'un-reviewed'
+        processed = 'processed'
+        failed = 'failed'
+
+    stage = db.Column(db.String(length=20), nullable=True)
```

Comment:
@karenc This feels fundamentally wrong but adding enums to tables via alembic seems to cause all sorts of migration problems. 
Are enums and alembic generally a bad combination. It seems like it at the mo. Any better suggestions?
This does seem to work at the moment but feels really clunky

Reply:
@hwindsor I looked into it and this is what I got:

```diff
commit 3c43a3b7f254208cbde0bee70f2bb4c570e4a865 (HEAD -> sighting-stage-num)
Author: karen chan <karen@karen-chan.com>
Date:   Wed Jun 9 14:45:26 2021 +0100

    Create sightingstage enum in migration explicitly
    
    Alembic didn't include anything to create the enum when it's "alter table" (I
    think), causing the type "sightingstage" to not exist:
    
    ```
    root@4b1ff276d7d2:/code# invoke app.db.upgrade --no-backup
    2021-06-09 12:54:45,315 [INFO] [app] Using app.config 'docker_local_config.LocalConfig'
    2021-06-09 12:54:46,012 [INFO] [alembic.runtime.migration] Context impl PostgresqlImpl.
    2021-06-09 12:54:46,013 [INFO] [alembic.runtime.migration] Will assume transactional DDL.
    2021-06-09 12:54:46,025 [INFO] [alembic.runtime.migration] Running upgrade fa952f537929 -> adb5d1a314bc, empty message
    2021-06-09 12:54:46,076 [INFO] [alembic.runtime.migration] Running upgrade adb5d1a314bc -> 58bf0e49cbf7, adding sighting stage
    2021-06-09 12:54:46,108 [CRITICAL] [tasks.app.db] Database upgrade failed
    Traceback (most recent call last):
      File "/usr/local/lib/python3.9/site-packages/sqlalchemy/engine/base.py", line 1276, in _execute_context
        self.dialect.do_execute(
      File "/usr/local/lib/python3.9/site-packages/sqlalchemy/engine/default.py", line 608, in do_execute
        cursor.execute(statement, parameters)
    psycopg2.errors.UndefinedObject: type "sightingstage" does not exist
    LINE 1: ALTER TABLE sighting ADD COLUMN stage sightingstage NOT NULL
                                                  ^
    ```
    
    After that, it complained about the column being null:
    
    ```
      File "/usr/local/lib/python3.9/site-packages/sqlalchemy/engine/base.py", line 1276, in _execute_context
        self.dialect.do_execute(
      File "/usr/local/lib/python3.9/site-packages/sqlalchemy/engine/default.py", line 609, in do_execute
        cursor.execute(statement, parameters)
    sqlalchemy.exc.IntegrityError: (psycopg2.errors.NotNullViolation) column "stage" contains null values
    
    [SQL: ALTER TABLE sighting ADD COLUMN stage sightingstage NOT NULL]
    ```
    
    And a plain `default='identification'` didn't work, had to use
    `server_default=sa.text("'identification'")` to insert some sql into "add
    column".
    
    See https://alembic.sqlalchemy.org/en/latest/ops.html

diff --git a/migrations/versions/58bf0e49cbf7_adding_sighting_stage.py b/migrations/versions/58bf0e49cbf7_adding_sighting_stage.py
index 087f1358..42fbbf25 100644
--- a/migrations/versions/58bf0e49cbf7_adding_sighting_stage.py
+++ b/migrations/versions/58bf0e49cbf7_adding_sighting_stage.py
@@ -25,8 +25,10 @@ def upgrade():
         ENTER DESCRIPTION HERE
     """
     # ### commands auto generated by Alembic - please adjust! ###
+    sightingstage = sa.Enum('identification', 'un_reviewed', 'processed', 'failed', name='sightingstage')
+    sightingstage.create(op.get_bind(), checkfirst=True)
     with op.batch_alter_table('sighting', schema=None) as batch_op:
-        batch_op.add_column(sa.Column('stage', sa.Enum('identification', 'un_reviewed', 'processed', 'failed', name='sightingstage'), nullable=False))
+        batch_op.add_column(sa.Column('stage', sightingstage, server_default=sa.text("'identification'"), nullable=False))
 
     # ### end Alembic commands ###
```

The tests passed: https://github.com/WildMeOrg/houston/runs/2784311137?check_suite_focus=true